### ISSUE TITLE
Multi select design/accessibility feedback

### ISSUE THREAD

**Issue by eseliger** [sourcegraph/sourcegraph#32560]
Collecting some thoughts I had when looking at the new MultiSelect component:

- Some hover styles are missing
  - The entire thing is clickable but has a regular cursor (maybe cannot change that)
  - The x-es next to items have a hover style, but:
  - The "remove all" x does not
  - The dropdown arrow does not either

- The dropdown looks confusing when I open it with three different background colors

Should selected elements just have a small checkmark indicator or something instead to not have so many colors at the same time, and especially two very similar blue tones?
<img width="830" alt="image" src="https://user-images.githubusercontent.com/19534377/158192710-13409989-db7b-46d5-892f-ce099de632b0.png">

- Accessibility: You can keyboard navigate to the "x"-es next to items, but pressing space or enter does nothing 
<img width="685" alt="image" src="https://user-images.githubusercontent.com/19534377/158193066-a0a0d26e-0d31-48d8-bd47-742d66425c95.png">

- The space above and below the list of elements in the dropdown looks weird when the elements are selected (I know that's the same it works on the dropdown of the navbar, but it is now more visible that the background color is blue and not light gray)
<img width="841" alt="image" src="https://user-images.githubusercontent.com/19534377/158193629-20654b0b-061b-4d90-aabe-2bf5c717ed90.png">

**Comment by oleggromov** [comment]
The PR addressed most of the listed issues so I am closing this one.
